### PR TITLE
Clarify and split up the "not actually SCTP" SCTP NetworkPolicy test

### PR DESCRIPTION
**Pull request by danwinship**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
The e2e suite has a test to make sure that plugins that don't implement SCTP nonetheless still behave in a sane way in the presence of NetworkPolices that mention SCTP. Unfortunately, people keep misunderstanding the point of the test and assuming that they can ignore it because they don't implement SCTP. (This has come up twice now with vendors certifying network plugins against OpenShift.)

Additionally, the existing test ("`should not allow access by TCP when a policy specifies only SCTP`") could fail for two different reasons, so I've now split it into two separate tests, with hopefully clearer descriptions:

* "`should not mistakenly treat 'protocol: SCTP' as 'protocol: TCP', even if the plugin doesn't support SCTP`"
  This tests that given a "deny all" policy and an "allow to SCTP port 81" policy, that the plugin does not allow traffic to *TCP* port 81. This is because some people seem to have implemented the "the default value for `protocol` is `TCP`" rule themselves, incorrectly, and end up treating everything that isn't `UDP` as `TCP`.
* "`should properly isolate pods that are selected by a policy allowing SCTP, even if the plugin doesn't support SCTP`"
  This tests that given _just_ an "allow to SCTP port 80" policy, that the plugin does not allow traffic to TCP port 81. That is, it tests that plugins still implement the "any pod selected by a NetworkPolicy is isolated" rule, even when the pod is selected by a NetworkPolicy they don't know how to implement.

I have three arguments for the latter test (beyond "this has always been part of the e2e suite so you always should have been passing this"):

1. NetworkPolicy is a security mechanism, so if you can't implement a policy _correctly_, then the user will probably be happier if you do something too restrictive than if you do something too permissive.
2. Except, you _can_ implement the policy correctly; if your plugin doesn't allow SCTP traffic between pods anyway, then you can implement a policy that says "don't allow anything except SCTP port 80" by just not allowing anything.
3. Based on NetworkPolicy semantics, users can expect that, eg, this pair of policies:

        kind: NetworkPolicy
        metadata:
          name: policy-tcp
        spec:
          ingress:
            - ports:
                - protocol: TCP
                  port: 80
        
        kind: NetworkPolicy
        metadata:
          name: policy-sctp
        spec:
          ingress:
            - ports:
                - protocol: SCTP
                  port: 80

    could correctly be squashed into a single policy:

        kind: NetworkPolicy
        metadata:
          name: policy-combined
        spec:
          ingress:
            - ports:
                - protocol: TCP
                  port: 80
            - ports:
                - protocol: SCTP
                  port: 80

    Any plugin that does not allow and deny exactly the same traffic given the combined policy as it would given the two separate policies is behaving inconsistently with the definition of NetworkPolicy.

#### Which issue(s) this PR fixes:
None

#### Does this PR introduce a user-facing change?
```release-note
Clarified the description of a test in the e2e suite that mentions "SCTP" but is
actually intended to be testing the behavior of network plugins that *don't*
implement SCTP.
```

/sig network
/priority important-longterm

### CODE REVIEW

File: test/e2e/network/netpol/network_policy.go [hunk 1]
```diff
@@ -1059,22 +1059,46 @@ var _ = common.SIGDescribe("Netpol", func() {
 			ValidateOrFail(k8s, model, &TestCase{ToPort: 80, Protocol: v1.ProtocolTCP, Reachability: denyIngressToXReachability})
 		})
 
-		ginkgo.It("should not allow access by TCP when a policy specifies only SCTP [Feature:NetworkPolicy] [Feature:SCTP]", func() {
```
Comment (knabben) [inline test/e2e/network/netpol/network_policy.go:1062]: We still have the old ones in place, should they be updated accordingly as well?
 https://github.com/kubernetes/kubernetes/blob/master/test/e2e/network/netpol/network_legacy.go#L1685

Comment (danwinship) [reply test/e2e/network/netpol/network_policy.go:1062]: ah, fixed

**Comment by knabben** [comment]
/lgtm

thanks for clarifying the tests.

**Pull request merged**